### ISSUE TITLE
FCM integration

### ISSUE THREAD

**Issue by zdnk** [customerio/customerio-expo-plugin#56]
Hello,

we have all tokens from FCM for iOS as well and we would like to import them in Customer.IO. However we also need to collect the new tokens as FCM tokens if I understand correctly because you cannot configure CustomerIO to send it as ASPN to some users as FCM to others right?

When is the integration planned?

**Comment by ami-aman** [comment]
Hi @zdnk 

Our Expo config plugin currently only support APNS for iOS and FCM for Android. 

We have _FCM integration_ request on our roadmap and have added your vote to the request. We will notify you when this is prioritized. Thanks for your feedback. We really appreciate it !

**Comment by zdnk** [comment]
What if I manually call CustomerIO.registerDeviceToken in javascript with the FCM token? Will that work?

**Comment by ami-aman** [comment]
I am afraid that would lead to device token mismatch i.e you will register FCM token using javascript and native module would generate an APNS token and register it.

**Comment by zdnk** [comment]
even if I have in my Expo project in the app.config.js disabled the notification registration?
```
ios: {
          useFrameworks: 'static',
          disableNotificationRegistration: true,
}
```

**Comment by ami-aman** [comment]
Yes, even if you have `disableNotificationRegistration :  true` then also this workaround might not work as expected. `AppDelegate.m` or `AppDelegate.mm` file would have delegate methods for APNS and if you attempt to send a FCM push notification, your device might never receive it. Let me know if you need more clarity on this. I would be happy to explain further.

**Comment by zdnk** [comment]
Ok. What can I do to make it work somehow? Because unfortunately we are using firebase and we have been using it with the old app version as well. I am open to suggestion where our backend register the devices in the customerio. But I need some solution :)

**Comment by ami-aman** [comment]
As a temporary solution, you could refer to the steps outlined in our React Native [documentation](https://customer.io/docs/sdk/react-native/push/). Ensure that you select FCM as you are using it for push notifications and choose either Objective-C or Swift based on your project. However, please note that the drawback of this workaround is that the changes made will only persist until you execute the 'prebuild --clean' command.

Do let us if this helps us. Feel free to comment if you need any assistance.

**Comment by zdnk** [comment]
Well, se are working with Expo managed for now. Using suggested solution by you does require bare workflow. 

What about adding option to simply completely disable push notification registration? This would actually solve the issue as it will allow developers to use FCM manually as workaround like I am doing right now.

**Comment by zdnk** [comment]
Any update here? Making it work together with Expo notifications would be also good.

**Comment by levibostian** [comment]
Adding FCM support to our Expo plugin is continuing to be tracked and prioritized internally. We do not have any updates to share at this time. We will update this issue when we have updates to share. 

Thank you, everyone!

**Comment by zdnk** [comment]
9 months later, whats the update?

**Comment by levibostian** [comment]
Thank you for following up on this feature request. We understand that waiting for updates can be frustrating, and we sincerely apologize for any inconvenience this may have caused.

Please know that we are continuing to track and prioritize this feature internally. Our team is dedicated to ensuring that we deliver high-quality updates, and this sometimes requires careful planning and evaluation to get things right.

Your feedback is incredibly important to us, and we truly appreciate your patience and understanding as we work through our development pipeline. 

Have a great day!

**Comment by Nikit-Singh** [comment]
Hey @levibostian, any updates on this?

**Comment by pablogdcr** [comment]
Also a blocker for us, please let us know if you have any updates on this!

**Comment by Shahroz16** [comment]
The support has been added `2.0.0` and onwards, thanks folks for the patience here and apologies that we forgot to update the thread.
